### PR TITLE
PUBDEV-5167 - pysparkling - reading parquet/orc data with time type m…

### PR DESCRIPTION
**Pull request by Pscheidl**
…essed up in H2O

### CODE REVIEW

**Comment by Pscheidl** [comment]
@michal-raska Please review requirements.txt

File: h2o-py/requirements.txt [hunk 1]
```diff
@@ -24,3 +24,4 @@ mysqlclient
 recommonmark
 sphinx_rtd_theme
 sphinxcontrib-osexample
+fastparquet
```
Comment (michalkurka) [inline h2o-py/requirements.txt]: In this specific case I would prefer to have the file statically available on S3. This is good for backwards compatibility tests.

And this is unrelated to the fact that fastparquet has dependencies that can conflict with our own :)

The right thing to do here IMHO is to create the file in Spark/Hive and use put it to S3.

Alternative solution is to write the file in junits.

Comment (mmalohlava) [reply h2o-py/requirements.txt]: Side note: requirements.txt should be used only for development. Actual package should use deps from `setup.py`.

Comment (michalkurka) [reply h2o-py/requirements.txt]: Very good point, thanks @mmalohlava - I did not catch that!

Comment (Pscheidl) [reply h2o-py/requirements.txt]: Thank you for your comments. I have resolved this issue. As soon as test parquet file is uploaded, I will push the changed requirements.txt and an updated test.

File: h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java [hunk 2]
```diff
@@ -180,6 +186,15 @@ public void addBinary(Binary value) {
...
       _writer.addNumCol(_colIdx, value, 0);
     }
+
+    @Override
+    public void addBinary(Binary value) {
+      final NanoTime nanoTime = NanoTime.fromBinary(value);
+      final long millisecondsToMidnight = DateTimeUtils.fromJulianDay(nanoTime.getJulianDay() - JULIAN_CALENDAR_INTEGER_ROUNDUP);
+      long millisecondsAfterMidnight =nanoTime.getTimeOfDayNanos() / NANOS_IN_MILLISECOND;
+      
+      _writer.addNumCol(_colIdx, millisecondsToMidnight + millisecondsAfterMidnight);
```
Comment (michalkurka) [inline h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java]: please also see the alternative implementation in Presto: https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTimestampUtils.java

what they are doing will be faster, but this is definitely fine as well

File: h2o-py/tests/testdir_jira/pyunit_pubdev_5167.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import h2o
+
+from tests import pyunit_utils
+
+
+def pubdev_5167():
+    parquet = h2o.import_file(path=pyunit_utils.locate("pubdev-5167.parquet"))
```
Comment (michalkurka) [inline h2o-py/tests/testdir_jira/pyunit_pubdev_5167.py]: smalldata/jira/
